### PR TITLE
Added ability to spawn individual selenium servers per build

### PR DESCRIPTION
**Pull request by job13er**
We were running into instances of selenium getting stuck and needing to be restarted, so we decided to just launch it for each build and then kill it after the tests.

### CODE REVIEW

File: src/exec.sh [hunk 1]
```diff
@@ -45,14 +52,27 @@ echo '-------------------------------'
...
-	echo "Running jasmine-node tests on port $TEST_PORT"
+    HTTP_PORT=$(getOpenPort)
+	kill $(lsof -t -i:$HTTP_PORT) 2>/dev/null || echo ''
+	./node_modules/.bin/http-server -s -c-1 -p $HTTP_PORT &
+
+    SELENIUM_PORT=$(getOpenPort)
+	kill $(lsof -t -i:$SELENIUM_PORT) 2>/dev/null || echo ''
+	./node_modules/.bin/webdriver-manager start --seleniumPort $SELENIUM_PORT > /dev/null 2>&1 &
+
+    ../bin/replace.js $TEST_CONFIG \
```
Comment (psbanka) [inline src/exec.sh]: This is a nice solution.

File: src/exec.sh [hunk 2]
```diff
@@ -62,18 +82,15 @@ function testIt {
 # start at the root of the project
 cd $DIR/..
 
-echo Processing $TARBALL...
-mkdir -p uploads/run
-mkdir -p screenshots
-mv build-$BUILD_INDEX build-${TIMESTAMP}
+echo Processing ${TARBALL}...
+mkdir build-${TIMESTAMP}
+ln -s ../build/node_modules build-${TIMESTAMP}/node_modules
```
Comment (psbanka) [inline src/exec.sh]: :+1: much better

File: src/exec.sh [hunk 3]
```diff
@@ -62,18 +82,15 @@ function testIt {
...
 tar -xzf ../uploads/$TARBALL
 testIt
 tar -cf ../screenshots/${TIMESTAMP}.tar spec/e2e/screenshots
 cd - # IN ROOT DIRECTORY ==================================
-rm -rf build-${TIMESTAMP}/demo
-rm -rf build-${TIMESTAMP}/spec
-mv build-$TIMESTAMP build-$BUILD_INDEX
+rm -rf build-${TIMESTAMP}
 
 if [[ $TEST_STATUS != 0 ]]; then exit $TEST_STATUS; fi
```
Comment (psbanka) [inline src/exec.sh]: this could just be `exit $TEST_STATUS`

**Comment by psbanka** [comment]
Looks great. One minor improvement that could be addressed in a future PR if you like

File: src/exec.sh (hunk 3 above)
Comment (job13er) [inline src/exec.sh]: Sure. I didn't write that line :wink:

**Comment by job13er** [comment]
@psbanka minor issue addressed

**Pull request merged**